doc/design/model_zoo.md

```diff
@@ -0,0 +1,174 @@
+
+### Model Zoo Data Schema
+
+The model zoo table is in a database deployed as part of the SQLFlow service. This database might not be the one that holds data sources.  The only requirement of the model zoo table is to have a particular data schema that contains at least the following fields.
+
+1. model ID (key), specified by the INTO clause, or `an_analyst/my_first_model` in the above example.
+1. Docker image ID, the Docker commit ID of the image `a_data_scientist/regressors` in the above example.
+1. submitter program, the source code of the submitter program, `my_first_model.py` in the above example, or its MD5 hash.
+1. data converter, the COLUMN and LABEL clauses.
+1. model parameter file path, the path to the trained model parameters on the distributed filesystem of the cluster.
+
+It is necessary to have the model ID so users can refer to the trained model when they want to use it.  Suppose that the user typed the prediction SQL statement at the beginning of this document. SQLFlow server will convert it into a submitter program and run it with the Docker image used to train the model. Therefore, the Docker image ID is also required. The model parameter path allows the prediction submitter program to locate and load the trained models.  The data converter helps the prediction submitter to use the conversion rules consistent with the ones used when training.
+
+It is necessary to record the content or the MD5 hash of the training submitter program in the model zoo table for experiment management. Please be aware that the training submitter encodes all three categories of hyperparameters, as listed in the above sections.  Suppose that the analyst re-trains the model with different hyperparameter settings, the training submitter changes accordingly, and SQLFlow should be able to remind the analyst to either uses a new model ID or overwrites the existing row in the model zoo table.
+
+### Model Sharing
+
+After all, what is a model zoo? A model zoo refers to all the model definitions and trained models accessible by a deployment of SQLFlow.  It contains one or more model definition Docker images and the source code repositories that build the images.  It also includes the model zoo table configured to work with all submitter programs generated by the deployment.
+
+Within a deployment, it is straightforward to share a trained model.  If the analyst, `an_analyst`, in the above example wants to use her own trained model `my_first_model` for prediction, she could use the short name `my_first_model`.
+
+```sql
+SELECT ... TO PREDICT ... USING my_first_model
+```
+
+If another analyst wants to use the trained model, he would need to use the full name as `an_analyst/my_first_model`.
+
+```sql
+SELECT ... TO PREDICT ... USING an_analyst/my_first_model
```

Comment:
I think using dot rather than slash  is more nature in SQL, such as "an_analyst.my_first_model".

Reply:
Both the full name of Docker images and trained models like the following are inspired by the convention used by Docker to name images.

- `dockerhub.com/a_data_scientist/regressors` or `a_data_scientist/regressors` names the Docker image hosted on `dockerhub.com`. This convention is from the Docker community.

- `models.sqlflow.com/an_analyst/my_first_model` or `an_analyst/my_first_model` names a model trained by `an_analyst` and hosted on `models.sqlflow.com`. I prefer to use the same convention of Docker image names, rather than inventing a new one like `an_analyst.my_first_model`.